src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -45,6 +45,17 @@ public CSharpType(Type type, bool isNullable, params CSharpType[] arguments)
             IsPublic = type.IsPublic && arguments.All(t => t.IsPublic);
         }
 
+        public CSharpType(Type type, bool isNullable, bool isValueType, params CSharpType[] arguments)
+        {
+            Debug.Assert(type.Namespace != null, "type.Namespace != null");
+            _type = type;
+
+            Namespace = type.Namespace;
+            Name = type.IsGenericType ? type.Name.Substring(0, type.Name.IndexOf('`')) : type.Name;
+            IsNullable = isNullable;
+            IsValueType = isValueType;
```

Comment:
If `CSharpType` is a wrapper around .net type, it can't be value type when .net type is a reference type and visa versa. These type properties must match.
What are you trying to achieve?